### ISSUE TITLE
Message Object Size

### ISSUE THREAD

**Issue by micahalcorn** [OriginProtocol/origin-dapp#487]
As [recommended by @crazybuster ](https://github.com/OriginProtocol/origin-dapp/issues/477#issuecomment-420434698), we should validate the size of the message object before storing it in OrbitDB since we are now supporting the inclusion of image data. I'm assuming that this should happen at the DApp layer since this is where we are doing the same for listings?

**Comment by franckc** [comment]
@tomlinton I believe on the IPFS node used for listings, upload limit is set to 2MB.
Is it the same on the messaging IPFS node ?

**Comment by franckc** [comment]
Sort of related to #489

**Comment by tomlinton** [comment]
It is the same for both at the moment I believe, it is enforced by the nginx proxy running in front of the IPFS nodes. The values might need to be changed to make them more suitable (at least for listings). The size limit needs to be enforced at the nginx proxy level because it will be easily circumvented in the DApp, but the DApp should also enforce it to provide better feedback/user experience.

I'm not totally across this so I might be wrong. I'm going to be looking into it over the next few days.

**Comment by micahalcorn** [comment]
This has become more of a priority because naive actors will upload images that will lock up their own browser and that of their recipients. A bad actor could easily spam and ruin every messaging-enabled DApp user's experience.

**Comment by tomlinton** [comment]
Because nginx limits the size you can't drop anything larger than 2mb into a message or it drops the connection. The user experience isn't great because it looks like the image was successfully sent and there are no warnings about any failures, but at least a malicious actor shouldn't be able to spam huge images.

Given that, do we still need the `p0` tag? There are a lot of improvements to be made here, but I'm not sure it is still blocking.

**Comment by DanielVF** [comment]
It's easy to increase nginx's limits.

**Comment by tomlinton** [comment]
For someone who wants to send a bunch of huge images? As far as I know there is no way to bypass the nginx setting.

**Comment by DanielVF** [comment]
We are talking about the maximum POST upload size in nginx?

client_max_body_size?

https://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size

**Comment by DanielVF** [comment]
That said, I think a 2mb limit is a good sane limit for image uploads.

**Comment by tomlinton** [comment]
You are right, it is easy to bypass. So it should only help with accidental cases.

**Comment by micahalcorn** [comment]
@tomlinton @DanielVF I'll remove the `p0` from this and create a separate issue for the case where someone circumvents the persistence layer and injects larger data.

**Comment by wanderingstan** [comment]
On the UX side, we really should be resizing images _in the browser_, so even naive users don't have to think about file sizes. E.g. they could upload a 10GB image and have it automatically scaled down to under 2MB. 

We already do this for the avatar image (via the npm package we're using). 

See the ancient issue #44 , especially [HTML5 image uploading](https://hacks.mozilla.org/2011/01/how-to-develop-a-html5-image-uploader/)

**Comment by joshfraser** [comment]
+1 for in-browser resizing
